File: scripts/fetch/test_sina_api.js
// test_sina_api.js - 测试新浪API能否获取完整历史
const https = require('https');
const http = require('http');
const fs = require('fs');

const symbol = process.argv[2] || 'sh510300';

function fetchSina(symbol, datalen) {
  return new Promise((resolve, reject) => {
    const url = `https://money.finance.sina.com.cn/quotes_service/api/json_v2.php/CN_MarketData.getKLineData?symbol=${symbol}&scale=240&ma=no&datalen=${datalen}`;
    console.log('请求URL:', url);
    
    https.get(url, {
      headers: {
        'Referer': 'https://finance.sina.com.cn/',
        'User-Agent': 'Mozilla/5.0'
      }
    }, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => resolve(data));
    }).on('error', reject);
  });
}

async function test() {
  console.log('=== 测试新浪API ===');
  console.log('代码:', symbol);
  console.log('');
  
  // 测试不同的datalen
  const testLens = [10, 100, 1000, 5000, 10000];
  
  for (const len of testLens) {
    console.log(`\n测试 datalen=${len}...`);
    try {
      const raw = await fetchSina(symbol, len);
      console.log('返回长度:', raw.length);
      
      if (raw.length > 0 && raw !== 'null') {
        const json = JSON.parse(raw);
        if (Array.isArray(json)) {
          console.log(`✅ 成功获取 ${json.length} 条`);
          if (json.length > 0) {
            console.log(`   最早: ${json[json.length-1].day}`);
            console.log(`   最新: ${json[0].day}`);
          }
          // 保存到文件
          fs.writeFileSync(`D:\\QClaw_Trading\\data\\test_sina_${symbol}_${len}.json`, JSON.stringify(json, null, 2));
        } else {
          console.log('⚠️ 返回不是数组:', typeof json);
        }
      } else {
        console.log('⚠️ 返回为空或null');
      }
    } catch(e) {
      console.log('❌ 失败:', e.message);
    }
    
    // 延迟
    await new Promise(r => setTimeout(r, 1000));
  }
}

test();

File: test_sina_api.py
#!/usr/bin/env python3
"""Test Sina daily kline limits."""
import requests, json

code = "512880"
for dl in [500, 1000, 1500, 2000, 3000, 5000]:
    url = f"https://quotes.sina.cn/cn/api/json_v2.php/CN_MarketData.getKLineData?symbol=sh{code}&scale=240&datalen={dl}"
    r = requests.get(url, timeout=15)
    if r.status_code == 200 and r.text.strip() and len(r.text) > 10:
        try:
            data = json.loads(r.text)
            if data:
                print(f"datalen={dl}: {len(data)} rec, first={data[0]['day']} last={data[-1]['day']}")
            else:
                print(f"datalen={dl}: empty")
        except:
            print(f"datalen={dl}: json err, text={r.text[:50]}")
    else:
        print(f"datalen={dl}: {r.status_code}, {len(r.text)} bytes, {r.text[:50]}")
